Container Stats problem on ARM/Raspberry pi

## Description

I'm running into problems trying to access the Docker stats via the Docker-Client on a Raspberry Pi (ARM architecture).  Seems like when I run docker.stats(<containerID>) it throws the error below.  I am able to call docker.info(), docker.listContainers(), so I know I can reach the Docker daemon.  

Interestingly, if I run this Docker container on Windows, I am able to call docker.stats() ok -- the cpu/memory stats show.



## How to reproduce

final DockerClient docker = DefaultDockerClient.fromEnv().build();
			Info info = docker.info();
			System.out.println("Docker info: " + info.toString());
			for (Container c : docker.listContainers()) {
				System.out.println("Container " + c.id());
				ContainerStats stats = docker.stats(c.id());
			}

I am running the docker container by exposing the socket (able to access docker from inside container): docker run -v /var/run/docker.sock:/var/run/docker.sock <container>

Btw, I am already rebuilding the libjffi library on the Raspberry Pi as mentioned in #477.  

## What do you expect

Expect to get a printout of the container stats.  

## What happened instead

It throws an exception below.

## Software:

- `docker version`: 

Client:
 Version:      17.04.0-ce
 API version:  1.28
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 18:22:23 2017
 OS/Arch:      linux/arm

Server:
 Version:      17.04.0-ce
 API version:  1.28 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 18:22:23 2017
 OS/Arch:      linux/arm
 Experimental: false

## Full backtrace

```text
Docker info: Info{architecture=armv7l, clusterStore=, cgroupDriver=cgroupfs, containers=48, containersRunning=2, containersStopped=46, containersPaused=0, cpuCfsPeriod=false, cpuCfsQuota=false, debug=false, dockerRootDir=/var/lib/docker, storageDriver=overlay2, driverStatus=[[Backing Filesystem, extfs], [Supports d_type, true], [Native Overlay Diff, false]], executionDriver=null, experimentalBuild=false, httpProxy=, httpsProxy=, id=N7RD:MR2K:JTJM:HM5B:MXWF:426E:3KCA:CXLF:JN44:BDGX:QSBZ:6EGB, ipv4Forwarding=true, images=158, indexServerAddress=https://index.docker.io/v1/, initPath=null, initSha1=null, kernelMemory=false, kernelVersion=4.1.7-v7+, labels=[], memTotal=970874880, memoryLimit=false, cpus=4, eventsListener=1, fileDescriptors=28, goroutines=43, name=raspberrypi, noProxy=, oomKillDisable=false, operatingSystem=Raspbian GNU/Linux 8 (jessie), osType=linux, plugins=Plugins{volumes=[], networks=[]}, registryConfig=RegistryConfig{indexConfigs={docker.io=IndexConfig{name=docker.io, mirrors=[], secure=true, official=true}}, insecureRegistryCidrs=[127.0.0.0/8]}, serverVersion=17.04.0-ce, swapLimit=false, systemStatus=[], systemTime=Mon May 15 21:36:42 UTC 2017}
Container 940ce42db20ed40a8b256659af9da3105a4c53db43411b6f13c251258b65e856
com.spotify.docker.client.exceptions.DockerRequestException: Request error: GET unix://localhost:80/containers/940ce42db20ed40a8b256659af9da3105a4c53db43411b6f13c251258b65e856/stats?stream=0: 200
	at com.spotify.docker.client.DefaultDockerClient.propagate(DefaultDockerClient.java:2261)
	at com.spotify.docker.client.DefaultDockerClient.request(DefaultDockerClient.java:2200)
	at com.spotify.docker.client.DefaultDockerClient.stats(DefaultDockerClient.java:1885)
	at monitor.Monitor.main(Monitor.java:56)
Caused by: javax.ws.rs.client.ResponseProcessingException: com.fasterxml.jackson.databind.JsonMappingException: Instantiation of [simple type, class com.spotify.docker.client.messages.MemoryStats] value failed: Null stats (through reference chain: com.spotify.docker.client.messages.ContainerStats["memory_stats"])
	at org.glassfish.jersey.client.JerseyInvocation.translate(JerseyInvocation.java:806)
	at org.glassfish.jersey.client.JerseyInvocation.access$700(JerseyInvocation.java:92)
	at org.glassfish.jersey.client.JerseyInvocation$5.completed(JerseyInvocation.java:773)
	at org.glassfish.jersey.client.ClientRuntime.processResponse(ClientRuntime.java:198)
	at org.glassfish.jersey.client.ClientRuntime.access$300(ClientRuntime.java:79)
	at org.glassfish.jersey.client.ClientRuntime$2.run(ClientRuntime.java:180)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:340)
	at org.glassfish.jersey.client.ClientRuntime$3.run(ClientRuntime.java:210)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Instantiation of [simple type, class com.spotify.docker.client.messages.MemoryStats] value failed: Null stats (through reference chain: com.spotify.docker.client.messages.ContainerStats["memory_stats"])
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.wrapException(StdValueInstantiator.java:399)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:231)
	at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:135)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:440)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1100)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:294)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:131)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:520)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:461)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:377)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1100)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:294)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:131)
	at com.fasterxml.jackson.databind.ObjectReader._bind(ObjectReader.java:1470)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:912)
	at com.fasterxml.jackson.jaxrs.base.ProviderBase.readFrom(ProviderBase.java:811)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.invokeReadFrom(ReaderInterceptorExecutor.java:256)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.aroundReadFrom(ReaderInterceptorExecutor.java:235)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:155)
	at org.glassfish.jersey.message.internal.MessageBodyFactory.readFrom(MessageBodyFactory.java:1085)
	at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:874)
	at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:808)
	at org.glassfish.jersey.client.ClientResponse.readEntity(ClientResponse.java:326)
	at org.glassfish.jersey.client.JerseyInvocation.translate(JerseyInvocation.java:803)
	... 17 more
Caused by: java.lang.NullPointerException: Null stats
	at com.spotify.docker.client.messages.AutoValue_MemoryStats.<init>(AutoValue_MemoryStats.java:24)
	at com.spotify.docker.client.messages.MemoryStats.create(MemoryStats.java:62)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.fasterxml.jackson.databind.introspect.AnnotatedMethod.call(AnnotatedMethod.java:120)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:227)
	... 39 more
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,13 @@
 # Changelog
 
+## 8.7.2
+
+### Bugfixes
+
+- Fix: make MemoryStat.stats nullable for ARM ([784][])
+
+[784]: https://github.com/spotify/docker-client/pull/784
+
 ## 8.7.1
 
 ### Bugfixes

--- a/src/main/java/com/spotify/docker/client/messages/MemoryStats.java
+++ b/src/main/java/com/spotify/docker/client/messages/MemoryStats.java
@@ -35,6 +35,7 @@ import javax.annotation.Nullable;
 @JsonAutoDetect(fieldVisibility = ANY, getterVisibility = NONE, setterVisibility = NONE)
 public abstract class MemoryStats {
 
+  @Nullable
   @JsonProperty("stats")
   public abstract Stats stats();
 

